### ISSUE TITLE
Dark mode doesn't work in Safari.

### ISSUE THREAD

**Issue by MalbaCato** [discordjs/guide#776]
**Describe the bug**
The dark mode button does nothing in Safari. An event handler triggers, but that seems to lead nowhere.

**To Reproduce**
Click the dark mode button. Observe that nothing happens.

**Device (please complete the following information):**
macOS 10.13.6
Safari 13.1.2, build (13609.3.5.1.5)

**Additional notes**
is there a deobfuscated version of the website files anywhere, just to help debug this?

**Comment by J-Human** [comment]
I can't repro but just FYI, the full "thing" that power's the guide's new theme is in here > https://github.com/discordjs/guide/tree/main/guide/.vuepress

**Comment by MalbaCato** [comment]
thanks, but that's not what I was asking for. I found where dark mode is referenced in the sources (just search for "dark" in the repo) but it doesn't help me much because it's not raw js and css, and I'm not familiar enough with vue nor scss to be able to statically analyse it.

**Comment by Danktuary** [comment]
From what I researched, I believe this is due to Safari not supporting `MediaQueryList#addEventListener`, which is being used on this line: https://github.com/discordjs/guide/blob/main/guide/.vuepress/theme/composables/useDarkMode.ts#L35
It looks like Safari 14+ supports it, though.

I don't have any devices I'd be able to debug this on, so I can't be sure, but I think a fix would be to use `MediaQueryList#onchange`, or maybe `MediaQueryList#addListener` (this is deprecated, though).

If you'd like to help debug this, clone the repo locally, `npm install`, `npm run dev`, and edit the line I linked above to use `mediaQuery.value.onchange = onMediaQueryChange` or `mediaQuery.value.addListener(onMediaQueryChange)`.
